chia/wallet/util/compute_hints.py

```diff
@@ -1,22 +1,54 @@
 from __future__ import annotations
 
-from typing import List
+from dataclasses import dataclass
+from typing import Iterator
 
-from chia.types.blockchain_format.program import INFINITE_COST
+from clvm.casts import int_from_bytes
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import INFINITE_COST, Program
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.coin_spend import CoinSpend
 from chia.types.condition_opcodes import ConditionOpcode
+from chia.util.ints import uint64
+
+
+@dataclass(frozen=True)
+class HintedCoin:
+    coin: Coin
+    hint: bytes32
+
+
+def hinted_coin_from_condition(condition: Program, parent_coin_id: bytes32) -> HintedCoin:
+    data = condition.as_python()
```

Comment:
@Quexington In your PR you moved away from using `Program.as_python` towards using `Program.at`. Im not sure if there was a deeper meaning to this and im not very familiar with all this `Program` stuff.. so, let me know.